lang/java/reef-io/src/main/java/org/apache/reef/io/network/NetworkServiceClient.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.network;
+
+import org.apache.reef.exception.evaluator.NetworkException;
+import org.apache.reef.io.network.impl.DefaultNetworkServiceImpl;
+import org.apache.reef.tang.annotations.DefaultImplementation;
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.Identifier;
+import org.apache.reef.wake.remote.Codec;
+import org.apache.reef.wake.remote.transport.LinkListener;
+
+import java.net.SocketAddress;
+
+@DefaultImplementation(DefaultNetworkServiceImpl.class)
+/**
+ * NetworkServiceClient.
+ * It can bind multiple ConnectionFactory.
+ */
+public interface NetworkServiceClient extends AutoCloseable {
+
+  /**
+   * Creates an instance of ConnectionFactory corresponding to the connectionFactoryId.
+   * Binds Codec, EventHandler, and LinkListener to the ConnectionFactory.
+   * @throws NetworkException throws a NetworkException when duplicated connectionFactoryId exists.
+   */
+  <T> void registerConnectionFactory(final Class<? extends Name<String>> connectionFactoryId,
```

Comment:
I followed `CommunicationGroupDriverImpl.addBroadcast` method. This method gets a `Class<? extends Name<String>> operatorName` parameter to register broadcast. 
Users can get this registered broadcast from `communicationGroupClient.getBroadcastSender(Class<? extends Name<String>> operatorName)` method. 

This could be `Identifier`. Is it better to use `Identifier`?